activerecord/lib/active_record/enum.rb

```diff
@@ -121,6 +121,13 @@ def inherited(base) # :nodoc:
       super
     end
 
+    def load_schema! # :nodoc:
+      attributes_to_define_after_schema_loads.each do |name, (cast_type, _default)|
+        cast_type = cast_type[type_for_attribute(name)] if Proc === cast_type
+        raise "Unknown attribute '#{name}' for #{self.name}" if Enum::EnumType === cast_type && !columns_hash.key?(name)
```

Comment:
```suggestion
        unless columns_hash.key?(name)
          cast_type = cast_type[type_for_attribute(name)] if Proc === cast_type
          raise UnknownAttributeError.new(self, name) if Enum::EnumType === cast_type
        end
```

A bit of a nitpick, but since we expect this not to happen, it may make sense to optimize for the columns to exist.

Reply:
Also maybe we should mention `enum` in the error message to make it easier to realize the source of the error is a faulty `enum` definition.